GUI/Util/SearchModifications.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Collections.ObjectModel;
+using System.Linq;
+using System.Windows.Controls;
+using System.Windows.Threading;
+
+namespace MetaMorpheusGUI
+{
+    class SearchModifications
+    {
+        public static DispatcherTimer Timer;
+        public static bool FixedSearch;
+        public static bool VariableSearch;
+        public static bool GptmdSearch;
+        
+        public static void SetUpModSearchBoxes()
+        {
+            Timer = new DispatcherTimer();
+            Timer.Interval = TimeSpan.FromMilliseconds(300);
+        }
+
+        // starts timer to keep track of user keystrokes
+        public static void SetTimer()
+        {
+            // Reset the timer
+            Timer.Stop();
+            Timer.Start();
+        }
+
+        // filters and expands tree according to user mod search
+        public static void FilterTree(TextBox textbox, TreeView tree, ObservableCollection<ModTypeForTreeView> collection)
+        {
+            string key = textbox.Text.ToLower();
+            if (string.IsNullOrEmpty(key))
+            {
+                tree.DataContext = collection; // shows full tree if nothing is searched
+                return;
+            }
+
+            var modTypesWithMatchingMods = collection.Where(p => p.Children.Any(c => c.DisplayName.ToLower().Contains(key))); // parent of child mods that match key
```

Comment:
i edited this region but it's still basically the same.

I'm wondering if it's faster to show/hide nodes in the tree rather than recreating a whole new one and copying over nodes from the old one to the new one. Can you try showing/hiding nodes in the tree rather than creating/setting a new one? I'm guessing it'll be more RAM efficient and faster but I've been reading that visibility of nodes in a treeview is hard to edit. plz check that out

Reply:
i figured out a way to do that, but it does add more code so maybe it's slower. each ModForTreeView would have a "Show" property. and because we'll be toggling between hide/show, we'll have to reset the collection if textbox is empty/null. what do u think?

Reply:
give it a shot and see if it's faster

Reply:
I do like this new approach better.

Reply:
feels about the same but can't really tell. the thing is the each mod type's children nodes are binded to a collection, so for this to work i'd still have to have create another collection for display purposes

Reply:
something like this:

public ObservableCollection<ModForTreeView> Children { get; }
```

        public ObservableCollection<ModForTreeView> DisplayChildren
        {
            get
            {
                var children = new ObservableCollection<ModForTreeView>();
                foreach (var child in Children)
                {
                    if (child.Show)
                    {
                        children.Add(child);
                    }
                }
                return children;
            }
        }
```